doc/rst/technotes/main.rst

```diff
@@ -36,89 +35,79 @@ For most uses, Chapel's existing config variable support is useful and
 easy; occasionally, however, one wants to fully control argument
 processing.
 
-If a program's main() function is declared to take no arguments, Chapel
-will behave as it traditionally has -- built-in flags and config
+If a program's main function is declared to take no arguments, Chapel
+will behave as it traditionally has: built-in flags and config
 settings will be handled and unrecognized flags will trigger an error.
 
-Alternatively, a main() function can be declared to take an array of
-strings as an argument.  By doing so, the program opts-in to explicit
-argument handling, described in the remainder of this README.
-
+Alternatively, ``main()`` can be declared to take an array of
+strings as an argument. By doing so, the program opts-in to explicit
+argument handling, described in the remainder of this technote.
 
 Usage
 -----
-To use this feature, declare main as follows:
 
-.. code-block:: chapel
+To use this feature, declare ``main()`` as follows:
 
-  proc main(args: [] string) {
-    // ...body...
-  }
+.. literalinclude:: ../../../test/technotes/doc-examples/MainProcExample0.chpl
+   :language: chapel
+   :start-after: START_EXAMPLE_0
+   :end-before: STOP_EXAMPLE_0
 
-When main is so declared, Chapel's default command line argument
+When ``main()`` is so declared, Chapel's default command line argument
 processing changes. Rather than generate an error message when an
 unknown argument is encountered, that argument will instead be passed
-on to this main(args) function. Arguments corresponding to config
+on to this ``main(args)`` function. Arguments corresponding to config
 variables or Chapel's predefined flags (like ``-nl``, etc.) will still
-be processed and will not be passed on to the main(args) function.
+be processed and will not be passed on to the ``main(args)`` function.
 
 To be clear, the arguments that would be handled without using this
 feature will continue to behave as usual. The difference is that other
-arguments will be passed to main(args) rather than generating an error
-message. The only exception to this, is ``--help`` and ``-h``.
+arguments will be passed to ``main(args)`` rather than generating an
+error message. The only exception to this, is ``--help`` and ``-h``.
 
 Prior to this feature, a Chapel program would always exit immediately
 after printing the help message. A programmer might want to print
 other information explaining what non-config-var command line
 arguments are available and what they might do. For this reason, when
-a Chapel program includes a main declared to take arguments, ``--help``
-and ``-h`` will be passed on to the main(args) function.  The function
-Help.printUsage() will print out the familiar config var table.
+a Chapel program includes a ``main()`` declared to take arguments,
+``--help`` and ``-h`` will be passed on to the ``main(args)`` function.
 
-To gain access to this new function requires a module:
+The function ``printUsage()`` will print out the familiar config var
```

Comment:
I bet we can make this a link to the actual function, though you might need a `:chpl:` tag before it to get the link to work

Reply:
Does that mean the syntax would be something like `:chpl: :proc:`printUsage()`? B/c I tried just `:proc:` thinking I'd be clever but the docs build didn't like it.

Reply:
The syntax looks to be
```
:chpl:proc:`ModName.symName`
```